src/display/text_layer.js

```diff
@@ -91,11 +146,10 @@ const renderTextLayer = (function renderTextLayerClosure() {
     }
     const fontHeight = Math.sqrt(tx[2] * tx[2] + tx[3] * tx[3]);
     let fontAscent = fontHeight;
-    if (style.ascent) {
-      fontAscent = style.ascent * fontAscent;
-    } else if (style.descent) {
-      fontAscent = (1 + style.descent) * fontAscent;
+    if (!style.ascent) {
+      style.ascent = getAscent(style.fontFamily, ctx);
     }
+    fontAscent *= style.ascent;
```

Comment:
If you want to, you can probably simplify this a little bit now that you always have a `style.ascent` value:
```javascript
    if (!style.ascent) {
      style.ascent = getAscent(style.fontFamily, ctx);
    }
    const fontAscent = fontHeight * style.ascent;
```